dotnet/src/Connectors/Connectors.OpenAI/AzureSdk/ClientCore.cs

```diff
@@ -623,9 +627,7 @@ await foreach (StreamingChatCompletionsUpdate update in response.ConfigureAwait(
                     // as the called function could in turn telling the model about itself as a possible candidate for invocation.
                     functionResult = (await function.InvokeAsync(kernel, functionArgs, cancellationToken: cancellationToken).ConfigureAwait(false)).GetValue<object>() ?? string.Empty;
                 }
-#pragma warning disable CA1031 // Do not catch general exception types
-                catch (Exception e)
-#pragma warning restore CA1031
+                catch (Exception e) when (!e.IsCriticalException())
```

Comment:
Not sure about this change, do we still need to use `IsCriticalException` check across the code?

Reply:
This is just a way to do general exception handling w/o the pragma.  If its a critical exception, the process is coming down.

Reply:
I remember there were some conversations about which exceptions should be considered as critical etc. Also, I'm not sure if this method is still up-to-date. Can we leave it as it was before (with `pragma`) and track the purpose of `IsCriticalException` as separate issue? Thanks!